lib/TwoFactorAuth.php

```diff
@@ -218,7 +207,8 @@ private function codeEquals(string $safe, string $user): bool
         // we don't leak information about the difference of the two strings.
         if (strlen($safe) === strlen($user)) {
             $result = 0;
-            for ($i = 0; $i < strlen($safe); $i++) {
+            $strlen = strlen($safe);
+            for ($i = 0; $i < $strlen; $i++) {
```

Comment:
I think separating the limit out to a variable makes this less readable

Reply:
I beg to differ, also recounting the strlen in every loop is resource intensive, depending on the input.